packages/nestjs-report/src/services/report-strategy.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Inject } from '@nestjs/common';
+import { ReportCreatableInterface } from '@concepta/ts-common';
+import { mapNonErrorToException } from '@concepta/ts-core';
+import { ReportGeneratorServiceNotFoundException } from '../exceptions/report-generator-service-not-found.exception';
+import { ReportTimeoutException } from '../exceptions/report-timeout.exception';
+import { ReportEntityInterface } from '../interfaces/report-entity.interface';
+import { ReportGeneratorResultInterface } from '../interfaces/report-generator-result.interface';
+import { ReportGeneratorServiceInterface } from '../interfaces/report-generator-service.interface';
+import { ReportSettingsInterface } from '../interfaces/report-settings.interface';
+import { ReportStrategyServiceInterface } from '../interfaces/report-strategy-service.interface';
+import { REPORT_MODULE_SETTINGS_TOKEN } from '../report.constants';
+
+export class ReportStrategyService implements ReportStrategyServiceInterface {
+  private readonly reportGeneratorServices: ReportGeneratorServiceInterface[] =
+    [];
+  constructor(
+    @Inject(REPORT_MODULE_SETTINGS_TOKEN)
+    private readonly settings: ReportSettingsInterface,
+  ) {}
+
+  public addStorageService(
+    reportGeneratorService: ReportGeneratorServiceInterface,
+  ): void {
+    this.reportGeneratorServices.push(reportGeneratorService);
+  }
+
+  async generate(
+    report: ReportEntityInterface,
```

Comment:
shound not be Entity

Reply:
it should be ReportCreatableInterface as well